### PR TITLE
Feature/create aircraft model

### PR DESCRIPTION
**Pull request by alexandreMelloTW**
:warning: _Is there something everyone needs to know before merging?_ (Optional)

### Requirement 
<details>
    <summary>
        <strong>Clearly explain your requirement.</strong> 
    </summary>
    <p>
       Implement an endpoint to create an Aircraft Model.
    </p>
</details>

### Solution
<details>
    <summary>
        <strong>Clearly explain your solution.</strong>           
    </summary>
    <p>
1. Created `Aircraft Model Controller` to receive an `Aircraft Model Form` and send it to `Aircraft Model Service`

2. Created `Aircraft Model Service` to _map_ `Aircraft Model Form` into `Aircraft Model` and save it by `Aircraft Model Repository`.

3. Created `Aircraft Model Repository` to handle _database persistence_.
    </p>
</details>

### How to test
<details>
    <summary>
        <strong>Request for test.</strong>
    </summary>
    <p>
curl -v -X POST http://localhost:8080/aircraft-model -H 'Content-Type: application/json' -d '{ "description": "my first model" }'
   </p>
</details>

### CODE REVIEW

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import br.com.caelum.clines.shared.domain.AircraftModel;
 import org.springframework.data.repository.Repository;
 
-import java.util.List;
-
 public interface AircraftModelRepository extends Repository<AircraftModel, Long>, ExistingAircraftModelService {
+    void save(AircraftModel aircraftModel);
```
Comment (MarceloLeite2604) [inline src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java:10]: Spring interface `Repository` already has a method `save(E entity)`. I don't think it is necessary redeclarate it.

File: src/test/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelFormMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package br.com.caelum.clines.api.aircraftmodels;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AircraftModelFormMapperTest {
+    private static final Long INPUT_AIRCRAFT_MODEL_ID = 1L;
```
Comment (lcsdtw) [inline src/test/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelFormMapperTest.java]: This one is not being used

**Review by lcsdtw — DISMISSED** [review]

**Review by FlaviaCarvalhoTW — DISMISSED** [review]

**Comment by lcsdtw** [comment]
Where is the mapper used? This PR won't be closed to create a branch feature/aircraft-model? Will this continue to be used to develop create aircraft model?

**Comment by alexandreMelloTW** [comment]
> Where is the mapper used? This PR won't be closed to create a branch feature/aircraft-model? Will this continue to be used to develop create aircraft model?

I'll fix that

**Review by lcsdtw — DISMISSED** [review]

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package br.com.caelum.clines.api.aircraftmodels;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotBlank;
+
+@Data
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
```
Comment (fwfurtado) [inline src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java:10]: Given that you have restricted the level of access in the constructor to the package

It makes sense to maintain the class's level of access as public?

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+public class AircraftModelForm {
+    @NotBlank
+    private String description;
+
+    AircraftModelForm(String description) {
```
Comment (fwfurtado) [inline src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java:15]: What do you think about of using `@AllArgsConstructor`?

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelService.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class AircraftModelService {
+    private final AircraftModelRepository repository;
+    private final AircraftModelFormMapper formMapper;
+
+    public Long createAircraftModelBy(AircraftModelForm form) {
+        var aircraftModel = formMapper.map(form);
```
Comment (fwfurtado) [inline src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelService.java:18]: It makes sense register two models with same description? Is there any way to validate this case?

File: src/test/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import br.com.caelum.clines.shared.domain.AircraftModel;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyObject;
```
Comment (fwfurtado) [inline src/test/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelServiceTest.java]: Is being used?

**Review by fwfurtado — DISMISSED** [review]

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java (hunk 4 above)
Comment (alexandreMelloTW) [reply src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java:15]: `AircraftModelForm` is instantiated only by Spring representing the request body. We can add it anytime if needed.

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java (hunk 3 above)
Comment (alexandreMelloTW) [reply src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java:10]: Any sugestion?

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java (hunk 1 above)
Comment (MarceloLeite2604) [reply src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java:10]: Previously answered by @fwfurtado on PR #6.

**Review by MarceloLeite2604 — APPROVED** [review]

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java (hunk 1 above)
Comment (alexandreMelloTW) [reply src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelRepository.java:10]: I just follow the pattern form `AircraftRepository`

**Review by FlaviaCarvalhoTW — APPROVED** [review]

File: src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java (hunk 3 above)
Comment (fwfurtado) [reply src/main/java/br/com/caelum/clines/api/aircraftmodels/AircraftModelForm.java:10]: Remove this restriction or apply this restriction to whole class in this package

**Pull request merged**